### ISSUE TITLE
Make cTeam use UUID's instead of player names

### ISSUE THREAD

**Issue by PureTryOut** [cuberite/cuberite#3234]
Currently, as explained by @tigerw [here](https://forum.cuberite.org/thread-2466-post-26041.html#pid26041), cTeam manages it's members by player names. As Minecraft currently uses UUID's, this should be used instead.

**Comment by PureTryOut** [comment]
A note, [I put a $10 bounty on this issue](https://www.bountysource.com/issues/35266687-make-cteam-use-uuid-s-instead-of-player-names) over on Bountysource.

Maybe something for you again @PlasmaPower?

**Comment by PlasmaPower** [comment]
Do I need to support backwards compatibility? If so, how?

**Comment by PureTryOut** [comment]
Keep names as an alternative you mean? Not sure why you would, Minecraft has been using UUID's for a long time now.

**Comment by PlasmaPower** [comment]
We don't want to break everything using the current API though.

**Comment by PureTryOut** [comment]
I honestly think there are 0 plugins out there right now using this, and even if plugins do use it, it's good for them to update to UUID's anyway, but sure make it backwards compatible. It will probably be a lot more work then though.

**Comment by PlasmaPower** [comment]
It won't be more work (I can just copy and paste). The reason not to is that then you have to add UUID to every function name. Also, plugin authors might gravitate towards the easier name based API. Maybe we should just replace the API then.

Anyone else have an opinion/would a huge breaking change get merged?

**Comment by Schwertspize** [comment]
UUID have a special structure right? You can check if it's a UUID or not.

**Comment by madmaxoft** [comment]
Actually, if I read things correct, Minecraft still uses player names for its team management, both in the commands and in the underlying storage. http://minecraft.gamepedia.com/Scoreboard , see the commands (`scoreboard players XYZ`) and the NBT structure (`/data/teams[]/players`)

Therefore I suggest we keep things as they are, but only improve the documentation so that it says what it uses.

If desired, we could add API functions that take UUIDs instead of player names, and internally translate to names. If Mojang ever decides to switch to UUIDs in team management, we could simply swap the internals and still support both kinds of functions.

**Comment by PureTryOut** [comment]
> If desired, we could add API functions that take UUIDs instead of player names, and internally translate to names.

I would prefer that then. I really dislike the idea of having to use player names where it asks for UUID's everywhere else. UUID's is more robust anyways, you don't get problems if players change their names.

**Comment by PlasmaPower** [comment]
@Schwertspize I'm not sure how that's relevant, older plugins won't be checking if it's a UUID and will therefore break with an update.

**Comment by Pokechu22** [comment]
Names are used for players, but entities in the scoreboard use their UUIDs (as strings).  (On the other hand, cuberite doesn't implement entity UUIDs fully yet, so it probably is a moot point)

**Comment by Schwertspize** [comment]
@PlasmaPower i mean if cuberite checks if the argument doesn't fit the UUID scheme, then assumes it's a name. 

Anyway, it doesn't have to be back backwards compatible

**Comment by madmaxoft** [comment]
@Schwertspize No, that's a really bad idea. Then a player comes along whose name is an uuid for some reason and all hell breaks lose. No, keep them separate and make separate functions.

**Comment by PlasmaPower** [comment]
@Schwertspize What about functions that take a team and return a UUID?

@madmaxoft not possible, MC usernames have a max of 16 chars, UUIDs are much longer.

**Comment by tigerw** [comment]
Please, none of this backwards compatibility stuff. Just change the names to the unique identifiers and deal with any complaints (probably none) which come. It's not worth the mess of having two functions.

Date: Sun, 26 Jun 2016 14:22:00 -0700

**Comment by madmaxoft** [comment]
It's not about backward compatibility, it's about ease of use. Sometimes it's easier to use a UUID (when you have a `cPlayer` object), sometimes it's easier to use a name (when handling a command with playername as param)

**Comment by PlasmaPower** [comment]
> when you have a cPlayer object

`Player.getUUID()`?

> It's perfectly worth having two functions

Operating on the same data set or different data sets? The first is very complicated.

**Comment by mathiascode** [comment]
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=35266687)](https://www.bountysource.com/issues/35266687-make-cteam-use-uuid-s-instead-of-player-names?utm_source=35266687&utm_medium=shield&utm_campaign=ISSUE_BADGE)
